src/css.js

```diff
@@ -255,8 +263,12 @@ jQuery.extend( {
 			if ( !hooks || !( "set" in hooks ) ||
 				( value = hooks.set( elem, value, extra ) ) !== undefined ) {
 
-				style[ name ] = value;
-			}
+                if ( customProperty ) {
+                    style.setProperty( name, value );
```

Comment:
I don't think we can use `setProperty` here because properties like `maxWidth` and `marginRight` don't work. I've delayed working on this because I keep running into this issue but according to the spec, and please correct me if I'm reading this wrong, camelcased properties aren't transformed into their CSS equivalents. Every test that fails when I switch to `setProperty` is precisely because of this.  

Sources: 
- https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-setproperty
- https://dom.spec.whatwg.org/#converted-to-ascii-lowercase